### PR TITLE
Update GKE ML blueprint to use native GKE driver

### PR DESCRIPTION
**Pull request by ankitkinra**
Updated blueprint to remove the custom installation of the GPU driver and instead use the GKE provided setting `gpu_driver_version` . Detailed list of changes

* Used the native config `gpu_driver_version`  present in `gpu_driver_installation_config` added in https://github.com/hashicorp/terraform-provider-google/issues/15151
* Removed `install-nvidia-drivers` which did the custom installation of drivers
* Removed dependency on `kubernetes-operations`

### CODE REVIEW

File: community/examples/ml-gke.yaml [hunk 1]
```diff
@@ -48,18 +48,18 @@ deployment_groups:
         cidr_block: $(vars.authorized_cidr)
     outputs: [instructions]
 
-  # Docs at https://github.com/GoogleCloudPlatform/hpc-toolkit/tree/main/community/modules/scripts/kubernetes-operations
```
Comment (nick-stroud) [inline community/examples/ml-gke.yaml:51]: I believe our documentation in examples/README.md may refer to this module, if so we should update that documentation.

Comment (nick-stroud) [inline community/examples/ml-gke.yaml:51]: Are there any other examples using the Kubernets-operations module. If not we should probably delete all of the module documentation in our Repo. This would be under community/modules/kubernetes-operations/ and also from modules/README.md

**Review by nick-stroud — CHANGES_REQUESTED** [review]

File: examples/README.md [hunk 2]
```diff
@@ -1382,10 +1382,10 @@ Toolkit. It includes:
 > work. See note below.
 
 * Creation of a regional GKE cluster.
-* Creation of an autoscaling GKE node pool with `a2` machines each with 8
-  attached A100 GPUs.
-* Configuration of the cluster using the [`kubernetes-operations`] module to
-  install nvidia drivers.
+* Creation of an autoscaling GKE node pool with `g2` machines each with 1
+  attached L4 GPUs. Note: We tested this blueprint with `a2` machines as well,
+  but as capacity is hard to find we are using `g2` for example at the moment.
```
Comment (nick-stroud) [inline examples/README.md:1387]: nit: I would reword to remove the word "we". AKA, "this blueprint was tested..."

File: community/examples/ml-gke.yaml [hunk 3]
```diff
@@ -19,6 +19,7 @@ vars:
   project_id:  ## Set GCP Project ID Here ##
   deployment_name: ml-01
   region: us-central1
+  zones: [us-central1-c]
```
Comment (nick-stroud) [inline community/examples/ml-gke.yaml:22]: Did you mean for this to make it in the final version, or is it an artifact from testing?

**Review by nick-stroud — CHANGES_REQUESTED** [review]

**Review by tpdownes — COMMENTED** [review]
This comment was stuck pending. I will see if it still applies with the changes.

File: community/examples/ml-gke.yaml (hunk 1 above)
Comment (ankitkinra) [reply community/examples/ml-gke.yaml:51]: Updated

Comment (ankitkinra) [reply community/examples/ml-gke.yaml:51]: Deleted references to `kubernetes-operations` module, there is only one reference remaining which is around an output variable which is required by the `kubernetes-operations`

Remaining lines which I plan to clean up in a separate CL unless we should do this together as well:
*  https://github.com/GoogleCloudPlatform/hpc-toolkit/blob/2af3c420f05970edb11f755af6556fde500b77d9/community/modules/scheduler/gke-cluster/README.md?plain=1#L159
* https://github.com/GoogleCloudPlatform/hpc-toolkit/blob/2af3c420f05970edb11f755af6556fde500b77d9/community/modules/scheduler/gke-cluster/outputs.tf#L23

File: community/examples/ml-gke.yaml [hunk 4]
```diff
@@ -48,18 +48,18 @@ deployment_groups:
...
-    use: [gke_cluster]
-    settings:
-      install_nvidia_driver: true
-
   - id: a2-pool
     source: community/modules/compute/gke-node-pool
     use: [gke_cluster]
     settings:
       machine_type: a2-highgpu-8g
+      guest_accelerator:
```
[the remark this answers is not in the document]
Comment (ankitkinra) [reply community/examples/ml-gke.yaml:58]: Should not apply anymore but I have added an example setting with a2 machines which has been tested once (but could not run job as I was not able to get capacity)

File: community/examples/ml-gke.yaml (hunk 3 above)
Comment (ankitkinra) [reply community/examples/ml-gke.yaml:22]: I can remove this as this was a change from a2 , when I was not able to get capacity , for g2 I probably don't need this .

Comment (ankitkinra) [reply community/examples/ml-gke.yaml]: Turns out I was wrong, `nvidia-l4` is not available in all zones so hardcoded to `us-central1-c` where it is present. Will add a comment to blueprint to record this reason.

File: community/examples/ml-gke.yaml [hunk 5]
```diff
@@ -19,6 +19,8 @@ vars:
   project_id:  ## Set GCP Project ID Here ##
   deployment_name: ml-01
   region: us-central1
```
Comment (tpdownes) [inline community/examples/ml-gke.yaml:21]: I suggest considering asia-southeast1-b in asia-southeast1 as a region with more capacity.

File: examples/README.md [hunk 6]
```diff
@@ -1382,10 +1382,28 @@ Toolkit. It includes:
 > work. See note below.
 
 * Creation of a regional GKE cluster.
-* Creation of an autoscaling GKE node pool with `a2` machines each with 8
-  attached A100 GPUs.
-* Configuration of the cluster using the [`kubernetes-operations`] module to
-  install nvidia drivers.
+* Creation of an autoscaling GKE node pool with `g2` machines each with 1
+  attached L4 GPUs. Note: This blueprint was tested this blueprint with `a2` machines
+  as well, but as capacity is hard to find we are using `g2` for example at the moment.
```
Comment (nick-stroud) [inline examples/README.md:1387]: Minor grammar errors, I recommend the following corrected version:

 Note: This blueprint has also been tested with `a2` machines, but as capacity is hard to find the example uses `g2` machines which have better obtainability. If using with `a2` machines it is recommended to first obtain an automatic reservation.

**Review by nick-stroud — APPROVED** [review]
